test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_get_info.cpp

```diff
@@ -0,0 +1,322 @@
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_NUM_QUEUES_KHR,
+            sizeof(ret_val) - 1, &ret_val, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return TEST_PASS;
+    }
+
+    cl_int RunQueuesInfoTest()
+    {
+        cl_uint num_queues = 0;
+
+        cl_int error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_NUM_QUEUES_KHR,
+            sizeof(num_queues) - 1, &num_queues, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        // cl_command_queue queues[number_of_queues];
+        std::vector<cl_command_queue> queues;
+        queues.resize(number_of_queues);
+
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_QUEUES_KHR,
+            number_of_queues * sizeof(cl_command_queue) - 1, queues.data(),
+            nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return TEST_PASS;
+    }
+
+    cl_int RunRefCountInfoTest()
+    {
+        cl_uint ref_count = 0;
+
+        cl_int error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR,
+            sizeof(ref_count) - 1, &ref_count, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+        return TEST_PASS;
+    }
+
+    cl_int RunStateInfoTest()
+    {
+        cl_uint state = 0;
+
+        cl_int error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_STATE_KHR, sizeof(state) - 1,
+            &state, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clGetCommandBufferInfoKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+        return TEST_PASS;
+    }
+
+    cl_int RunPropArrayInfoTest()
+    {
+        std::vector<cl_command_buffer_properties_khr> properties;
+        properties.resize(number_of_properties);
+
+        cl_int error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR,
+            number_of_properties * sizeof(cl_command_buffer_properties_khr) - 1,
```

Comment:
The result of this calculation is:
`0 * sizeof(cl_command_buffer_properties_khr) - 1` which is `-1`, but this is a `size_t`. This test is incorrect.